packages/shared/src/contexts/AuthContext.tsx

```diff
@@ -78,8 +78,8 @@ export const getQueryParams = (): Record<string, string> => {
 
 export const REGISTRATION_PATH = '/register';
 
-const logout = async (): Promise<void> => {
-  await dispatchLogout();
+const logout = async (reason?: string): Promise<void> => {
```

Comment:
Should we make reason a required parameter? That way, the typescript type system will enforce having the parameter everywhere the logout function is used. Otherwise we might potentially miss places where we should have added it and didn't.

Reply:
Probably could make it required, it'll fallback on the API side anyways. There are two other places it is used currently

This one in AuthModal, not really sure if it's even in use anymore. Don't think you ever see the AuthModal anymore if you're logged in.
https://github.com/dailydotdev/apps/blob/345db939e85fcff112b29da2c67c9f62e34f7297/packages/shared/src/components/auth/AuthModal.tsx#L37-L39

And this one in AuthContext. Don't fully understand when this would be triggered, haven't looked too much into it
https://github.com/dailydotdev/apps/blob/345db939e85fcff112b29da2c67c9f62e34f7297/packages/shared/src/contexts/AuthContext.tsx#L137-L139

Maybe @rebelchris has some insight to those two.

Reply:
Both are basically for users who start onboarding but never finish.
The last one specifically for when they reload and didn't finish signing up. 

I'm ok with making it required, not many implementations anyway.

Reply:
Made it required and added a new reason `IncomleteOnboarding = 'incomplete onboarding',`